### ISSUE TITLE
[REQUEST] Variant of the range highlighter that takes highlighter paths instead of face names

### ISSUE THREAD

**Issue by sclu1034** [mawww/kakoune#4826]
### Feature

A version of the `range` highlighter where the string part of the `range-specs` is interpreted as a highlighter name/path instead of a face name.
The pointed to highlighter is then applied to the region covered by the entry's start and end position.

### Usecase

The ability to highlight a file that comprises multiple regions covered by existing highlighters, but where no delimiters exist in the text to apply a regex-based region highlighter to.

One example would be rendering Markdown (i.e. removing the Markdown syntax, but keeping the highlighting) with code snippets into a buffer: https://github.com/kak-lsp/kak-lsp/issues/576.

**Comment by SeerLite** [comment]
Yes please! Many times I've come across a string in a language X that contains source code for another language Y. I'd love to be able to highlight the specific string temporarily (e.g. by selecting the content, adding a highlighter and then removing it after I'm done) .

```nix
example = {
    enable = true;
    scriptContent = ''
        import os
        print(os.getcwd().replace("/", " "))
    '';
};
```
`<a-i> i :highlight python <ret>`

```python
example = {
    enable = true;
    scriptContent = ''
        import os
        print(os.getcwd().replace("/", " "))
    '';
};
```

(But without losing the nix language highlighting)
I wouldn't use region with `''` delimiter here because there may be other `''` multiline strings in the file that I don't want to see as Python. Also the delimiter may not be obvious in languages like YAML.

**Comment by mawww** [comment]
I do not remember seeing this issue back when it was opened, I must have not been paying enough attention, this is actually a great idea that might be generalized.

We already have the concept of highlighter delegation in the regions system, that delegates to other sub highlighters, it would make a lot of sense to generalize this by so that a "ranges", "regex" and possibly "line"/"column" highlighter just delegates to another one (using the "fill" highlighter to manitain the current behaviour).

The main issue with such a change is that it would be a breaking change to use the obvious syntax for this (just writing the delegate highlighter), but I think we might keep backwards compatibility by treating "single word" as a face and "multiple words" as a delegate highlighter spec.
